coldfront/plugins/qumulo/tests/test_tasks.py

```diff
@@ -245,3 +261,256 @@ def test_conditionally_update_storage_allocation_statuses_checks_all_pending_all
+        self, qumulo_api_mock: MagicMock
+    ) -> None:
+        qumulo_api = MagicMock()
+        qumulo_api.get_all_quotas_with_usage.return_value = self.mock_quota_response
+        qumulo_api_mock.return_value = qumulo_api
+
+        try:
+            ingest_quotas_with_daily_usage()
+        except:
+            self.fail("Ingest quotas raised exception")
+
+        base_quotas = get_mock_quota_base_allocations(self.STORAGE2_PATH)
+        for path, quoata_data in base_quotas.items():
+            storage_filesystem_path_attribute = AllocationAttribute.objects.select_related(
+                "allocation"
+            ).get(
+                value=path.rstrip("/"),
+                allocation_attribute_type=self.storage_filesystem_path_attribute_type,
+                allocation__status=self.status_active,
+            )
+
+            allocation = storage_filesystem_path_attribute.allocation
+            storage_quota_attribute = AllocationAttribute.objects.get(
+                allocation=allocation,
+                allocation_attribute_type=self.storage_quota_attribute_type,
+            )
+
+            allocation_attribute_usage = (
+                storage_quota_attribute.allocationattributeusage
+            )
+
+            usage = int(quoata_data["usage"])
+            self.assertEqual(allocation_attribute_usage.value, usage)
+            self.assertEqual(allocation_attribute_usage.history.first().value, usage)
+            self.assertGreater(allocation_attribute_usage.history.count(), 1)
+
+    @patch("coldfront.plugins.qumulo.tasks.QumuloAPI")
+    def test_doesnt_ingest_sub_allocation_data(
+        self, qumulo_api_mock: MagicMock
+    ) -> None:
+        qumulo_api = MagicMock()
+        qumulo_api.get_all_quotas_with_usage.return_value = self.mock_quota_response
+        qumulo_api_mock.return_value = qumulo_api
+
+        try:
+            ingest_quotas_with_daily_usage()
+        except:
+            self.fail("Ingest quotas raised exception")
+
+        sub_quotas = get_mock_quota_sub_allocations(self.STORAGE2_PATH)
+        for path in sub_quotas.keys():
+            allocation_attribute_usage = None
+            storage_filesystem_path_attribute = AllocationAttribute.objects.get(
+                value=path.rstrip("/"),
+                allocation_attribute_type=self.storage_filesystem_path_attribute_type,
+            )
+
+            allocation = storage_filesystem_path_attribute.allocation
+            storage_quota_attribute = AllocationAttribute.objects.get(
+                allocation=allocation,
+                allocation_attribute_type=self.storage_quota_attribute_type,
+            )
+
+            allocation_attribute_usage = (
+                storage_quota_attribute.allocationattributeusage
+            )
+
+            self.assertEqual(allocation_attribute_usage.value, 0)
+            self.assertEqual(allocation_attribute_usage.history.first().value, 0)
```

Comment:
@harterjwustl Perhaps it should be more appropriate to check that the history should always have one record.

```
            self.assertEqual(allocation_attribute_usage.history.count(), 1)
```